shibboleth-integration-idpv3/src/aaf/vhr/idp/http/VhrRemoteUserAuthServlet.java

```diff
@@ -237,7 +263,36 @@ protected void service(final HttpServletRequest httpRequest, final HttpServletRe
                 httpRequest.setAttribute(ExternalAuthentication.AUTHENTICATION_INSTANT_KEY, authnInstantArr[0]);
             };
 
-            httpRequest.setAttribute(ExternalAuthentication.PRINCIPAL_NAME_KEY, username);
+            final Subject subject = new Subject();
+            subject.getPrincipals().add(new UsernamePrincipal(username));
+
+            // If mfaPrincipalName is configured, add the correct set of principals to Subject.
+            // The principal name matching MFA will only be added if MFA was used.
+            // Other principal names supported by this flow are ssumed to be SFA and will be added regardless of MFA.
```

Comment:
Nit: "ssumed" -> "assumed"

Reply:
Thanks for the careful read - fixed in 2f587b3